packages/headless/src/test/mock-search-request.ts

```diff
@@ -26,6 +26,10 @@ export function buildMockSearchRequest(
     accessToken: '',
     visitorId: '',
     debug: false,
+    filterField: undefined!,
+    parentField: undefined!,
+    childField: undefined!,
+    filterFieldRange: undefined!,
```

Comment:
Why not set these values to empty strings or `0` in the case of `filterFieldRange`?

Reply:
Those fields are only included in the request when we intend to do folding. Adding them to the mock causes some tests to fail, since they don't include those fields.

Reply:
I see, it's not clean to work-around the problem this way though, overriding typescript strict null check. Also, if a developer tries to access `filterFieldRange`, they'll get `undefined`, when typescript would say it is a number.

I'm curious to know, which tests are failing?

Reply:
Hmm, why don't we send everything instead to keep things simple? When I look at `buildSearchRequest`, most parts of the request are not always sent. Why should `folding` be an exception?

Reply:
you're right, I did more research and it turns out that `buildMockSearchRequest` is only used by facet tests, and it includes all those fields because they happen to be included in facet search requests. I'm going to remove this mock method and use `buildSearchRequest` in unit tests instead.